state/core_access.go

```diff
@@ -256,6 +290,10 @@ func (ca *CoreAccessor) SubmitPayForBlob(
 		if response != nil && response.Code != 0 {
 			err = errors.Join(err, sdkErrors.ABCIError(response.Codespace, response.Code, response.Logs.String()))
 		}
+
+		if err != nil && errors.Is(err, sdkerrors.ErrNotFound) && !ca.granter.Empty() {
+			return response, errors.New("granter has revoked the grant")
```

Comment:
I wouldn't do a super complicated err check for this unless the error returned from app is *so* unclear that we need to do this check

Reply:
we are getting back "not found" which is very confusing.